dts/bindings/wifi/nordic,nrf7002-qspi.yaml

```diff
@@ -0,0 +1,85 @@
+# Copyright (c) 2022 Nordic Semiconductor ASA
+# SPDX-License-Identifier: Apache-2.0
+
+
+description: This is a representation of the nRF7002 WiFi chip.
+
+compatible: "nordic,nrf7002-qspi"
+
+include: "nordic,nrf7002.yaml"
+
+on-bus: qspi
+
+properties:
+  sck-frequency:
+    type: int
+    required: true
+    description: |
+      Maximum clock speed supported by the device, in Hz.
+
+  reg:
+    required: true
+
+  label:
+    required: false
+
+  readoc:
+    type: string
+    required: false
+    enum:
+      - "fastread"        # Single data line SPI, FAST_READ (0x0B)
+      - "read2o"          # Dual data line SPI, READ2O (0x3B)
+      - "read2io"         # Dual data line SPI, READ2IO (0xBB)
+      - "read4o"          # Quad data line SPI, READ4O (0x6B)
+      - "read4io"         # Quad data line SPI, READ4IO (0xEB)
+    description: |
+      Specify the number of data lines and opcode used for reading.
+      If not provided fastread will be selected.
+
+  writeoc:
+    type: string
+    required: false
+    enum:
+      - "pp"            # Single data line SPI, PP (0x02)
+      - "pp2o"          # Dual data line SPI, PP2O (0xA2)
+      - "pp4o"          # Quad data line SPI, PP4O (0x32)
+      - "pp4io"         # Quad data line SPI, PP4IO (0x38)
+    description: |
+      Specify the number of data lines and opcode used for writing.
+      If not provided pp will be selected.
+
+  address-size-32:
+    type: boolean
+    required: false
+    description: |
+      Set to indicate that 32-bit addressing is to be used.
+      If not specified 24-bit addressing will be used.
+
+  sck-delay:
+    type: int
+    required: false
+    default: 0
```

Comment:
You must justify **why** you have chosen this default value in the description.

Reply:
IMHO we can leave this as is, feel free to re-open if we can find exact reason for having a default here (as its copied from existing file).

Reply:
> IMHO we can leave this as is

Unresolving as this is not a matter of opinion. Here is the list of rules for upstream Zephyr devicetree bindings:

https://docs.zephyrproject.org/latest/build/dts/bindings.html#rules-for-mainline-bindings

Note this rule: 

> All recommendations in [Default values for properties](https://docs.zephyrproject.org/latest/build/dts/bindings.html#dt-bindings-default) are requirements when submitting the binding.
>
> In particular, if you use the default: feature, you must justify the value in the property’s description.

NCS bindings follow these same rules.

However, what is very confusing is that this `sck-delay` property is being added in this patch but is then removed in a subsequent fixup commit, leaving the entire situation unclear. What is going on?

Reply:
After offline discussion with @anangl  we have cleaned up QSPI bindings, as part of that this is now removed and instead we are hardcoding in the driver as per suggestions. So, closing this now.